models/result.go

```diff
@@ -49,7 +55,49 @@ func (r *Result) createEvent(status string, details interface{}) (*Event, error)
 		if err != nil {
 			return nil, err
 		}
-		e.Details = string(dj)
+
+		if EVENT_DATA_SUBMIT == status && c.PublicKeyId != 0 { // Zero is unset
+			//Taken from crypto/cipher CFB example
```

Comment:
I'll have to review this a bit more closely. Tbh, I'm not a crypto expert so I'll take a look at the examples provided in the Golang source that you mentioned.

Reply:
Also, let's move this to a function in `publickey.go`. Perhaps creating a function like this (untested):
```suggestion
			func (pk *PublicKey) Encrypt(data []byte) ([]byte, error) {} 
```

Reply:
That's a good suggestion. As for the crypto side of things, if you'd like me to do a bit of a walk through of my thought process for it I'd be more than happy. 
I would suggest reading through the source first, so you spot anything I might bias you against with an explanation.